### ISSUE TITLE
Exploding gradients with Matlab2015a+Cuda7.0+Cudnn4 in MatConvNet-beta18

### ISSUE THREAD

**Issue by nightrome** [vlfeat/matconvnet#401]
After upgrading from beta17 to beta18 and using the new provided models, I experience exploding gradients (and eventually nan and inf values) on both AlexNet and VGG16 (with and without response normalization layer). This was not the case in beta17. And it only occurs when I compile with Cudnn support. Otherwise there are no problems, despite still using the GPU.
Are there any known incompatibilities with Matlab2015a+Cuda7.0+Cudnn4 in MatConvNet-beta18 on Ubuntu 15.04?

**Comment by vedaldi** [comment]
Hi, not that we are aware of, but this is quite interesting info. I did notice a few minor bugs in cuDNN but they did not affect our results.  It is possible that the bug occurs only on certain GPUs as cuDNN uses different code paths for different architectures. What GPU are you using?

**Comment by nightrome** [comment]
I am using (several) GTX Titan X.
It turns out that the problem also exists - although to a lesser degree - without Cudnn on the new imagenet-matconvnet-alex model, but not on imagenet-caffe-alex, imagenet-caffe-ref or imagenet-matconvnet-vgg-verydeep-16 models.

**Comment by nicjac** [comment]
I have the exact same setup (except on Windows and with a single Titan X)
and I noticed some strange behaviours during training, especially when
batch normalization is used. For example, the testing error would suddenly
spike at values that are one or two orders of magnitude larger than
expected. I never noticed this in beta 17 and the networks I train are
pretty much the same. Could it be caused by batch normalization somehow? I
used to fix the parameters by hand but I now let MatConvNet handle it.

Nicolas

**Comment by vedaldi** [comment]
Hi, thanks for reporting. We are now investigating to see if we can reproduce the problem.
